Scripts/Cast/FixedLineCast.cs

```diff
@@ -0,0 +1,45 @@
+﻿namespace VRTK.Core.Cast
+{
+    using UnityEngine;
+
+    /// <summary>
+    /// Casts a straight line in the direction of the origin for a fixed length.
+    /// </summary>
+    public class FixedLineCast : StraightLineCast
+    {
+        /// <summary>
+        /// The current length of the cast.
+        /// </summary>
+        [Tooltip("The current length of the cast.")]
+        public float currentLength = 1f;
+
+        /// <summary>
+        /// Sets the length of the cast to the given length.
+        /// </summary>
+        /// <param name="length">The new length of the cast.</param>
+        public virtual void SetLength(float currentLength)

```

Comment:
Should this be called `SetCurrentLength` instead?

I see why the field isn't just `length` (potential confusion with the superclass' length field), but these setter methods should probably be using the same naming compared to the field.